### PR TITLE
[#9750] Instructor: Courses/Enroll: Limit of 100 students per section is not enforced

### LINKED ISSUE #9747: Instructor: Courses/Enroll: Unable to update Section name of student
- **Environment**: 

`master` branch at commit 1302ea06ea723385d310e4d844bab669cd931859

**Steps to reproduce**
1. Navigate to /instructor/courses/enroll?courseid=ANY_VALID_COURSE_ID
2. Ensure that it has at least one student and their **Team** name is **unique** in the course. e.g:

Section | Team | Name | Email
-- | -- | -- | --
A | Team 1 | Some name | Email@email.com

3. In the enroll page, under **New Students**, try to modify the existing student's section name. e.g:

Section | Team | Name | Email
-- | -- | -- | --
B | Team 1 | Some name | Email@email.com

**Expected behaviour**
The student's section name will be updated to "B".

**Actual behaviour**
Observe the error message below. This should not be an error because sections A and B do not exist concurrently and what we're trying to do here is to transfer Team 1 from section A to B.
image.

**Additional info**
Editing the section name through the **Edit Student Details** page works unlike the above.

### LINKED ISSUE #9750: Instructor: Courses/Enroll: Limit of 100 students per section is not enforced
- **Environment**: 
`master` branch at commit 4346f6f3d5ef8bf9570ab088a9835709f13ca99f

**Steps to reproduce**
1. Log in as instructor
2. Go to `/instructor/courses/enroll?courseid=ANY_VALID_COURSE_ID`
3. Add more than 100 students in a single section

**Expected behaviour**
Should not be allowed.

**Actual behaviour**
Students are enrolled and one section contains more than 100 students.
image

**Additional info**
When editing a single student through the  `/instructor/courses/student/edit?` page into a section with 101 >= students (this is another bug. It should be 100 and not 101. See #9752), an error will be displayed.
image
This should be the expected behaviour to the issue stated above.

### PR DESCRIPTION
**Pull request by erjillsison**
Fixes #9750 
Fixes #9747

**Outline of Solution**

- Make use of existing `logic.validateSectionsAndTeams()` in `EnrollStudentsAction` to check for section limit and team name validity.
- Update error messages and place them in a new nested class `StudentsLogicConst` in `Const`

### CODE REVIEW

**Comment by erjillsison** [comment]
Ready for review

File: src/test/java/teammates/test/cases/webapi/EnrollStudentsActionTest.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public void testExecute_withDuplicatedTeamNameAmongSectionsToExistingStudents_sh
         student1.section = "random section 1";
         StudentsEnrollRequest req = prepareRequest(Arrays.asList(student1));
         loginAsInstructor(typicalBundle.instructors.get("instructor1OfCourse1").getGoogleId());
-        verifyDuplicatedTeamNameDetected(courseId, req, student1.team, studentInCourse1.section, student1.section);
+        verifyDuplicatedTeamNameDetected(courseId, req, student1.team, student1.section, studentInCourse1.section);
```
Comment (Adoby7) [inline src/test/java/teammates/test/cases/webapi/EnrollStudentsActionTest.java]: You need to add test cases at least for the two issues you solved

File: src/main/java/teammates/logic/core/StudentsLogic.java [hunk 2]
```diff
@@ -25,8 +26,34 @@
  * @see StudentsDb
  */
 public final class StudentsLogic {
+    /**
+     * Error message when trying to create the same team in more than one section.
+     */
+    public static final String ERROR_INVALID_TEAM_NAME =
```
[the remark this answers is not in the document]
Comment (erjillsison) [reply src/main/java/teammates/logic/core/StudentsLogic.java]: Roger. Moved them to a new class in `Const`.

File: src/test/java/teammates/test/cases/webapi/EnrollStudentsActionTest.java (hunk 1 above)
Comment (erjillsison) [reply src/test/java/teammates/test/cases/webapi/EnrollStudentsActionTest.java]: Updated with new test cases

File: src/main/java/teammates/logic/core/StudentsLogic.java [hunk 3]
```diff
@@ -286,27 +284,35 @@ private String getSectionInvalidityInfo(List<StudentAttributes> mergedList) {
...
 
         return errorMessage.toString();
     }
 
     private String getTeamInvalidityInfo(List<StudentAttributes> mergedList) {
-
+        StringJoiner errorMessage = new StringJoiner(" ");
         StudentAttributes.sortByTeamName(mergedList);
 
         List<String> invalidTeamList = new ArrayList<>();
```
Comment (Adoby7) [inline src/main/java/teammates/logic/core/StudentsLogic.java]: It seems that using HashMap will improve the performance, but I am OK with it now.

**Pull request merged**